java/pom.xml

```diff
@@ -314,6 +314,7 @@
           </executions>
           <configuration>
             <failOnWarning>true</failOnWarning>
+            <ignoreNonCompile>true</ignoreNonCompile>
```

Comment:
Do we need this?

Reply:
I believe so - the maven dependency plugin gives warnings for test-scoped dependencies not used in main (which causes packaging to fail) without this